ampd/src/evm/verifier.rs

```diff
@@ -17,7 +18,9 @@ impl PartialEq<IAxelarGatewayEventsWithLog<'_>> for &Message {
             IAxelarAmplifierGatewayEvents::ContractCallFilter(event) => {
                 log.transaction_hash == Some(self.message_id.tx_hash.into())
                     && event.sender == self.source_address
-                    && self.destination_chain == event.destination_chain
+                    && self.destination_chain
+                        == ChainName::try_from(event.destination_chain.as_ref())
+                            .expect("failed to parse to ChainName")
```

Comment:
user provides this value in the event, so the conversion to ChainName can fail. If the conversion, then this should automatically return false. It might be useful to have a debug log and print the conversion error too

Reply:
agree with both points (no expect + add logging in case of conversion failure)